### ISSUE TITLE
Distribution size control?

### ISSUE THREAD

**Issue by donn** [KLayout/klayout#1005]
So I've recently been trying to compile Klayout from scratch. I used a script that's a modified version of your `makedeb.sh`. Problem is, while your deb file is a reasonable 54.6 MB compressed, mine is a far less reasonable ~~621 GB~~ 621 MB compressed.

Is there something I'm missing?

```sh
#!/bin/bash
set -e
set -x

bininstdir="./tmp/bin" 
builddir="./tmp/build" 

# destination folders
bindir="/build/bin"
libdir="/build/lib/klayout"

distpackdir="/build/lib/python3/dist-packages"
pylibdir="$distpackdir/klayout"

# clean bin directory
rm -rf $bininstdir

# do the actual build
./build.sh -j$(nproc) \
           -bin $bininstdir \
           -build $builddir \
           -rpath $libdir 

echo "Copying files .."

mkdir -p ${libdir}/db_plugins
mkdir -p ${libdir}/lay_plugins
mkdir -p ${pylibdir}
mkdir -p ${bindir}

cp -pd $bininstdir/strm* ${bindir}
cp -pd $bininstdir/klayout ${bindir}
cp -pd $bininstdir/lib*so* ${libdir}
cp -pd $bininstdir/db_plugins/lib*so* ${libdir}/db_plugins
cp -pd $bininstdir/lay_plugins/lib*so* ${libdir}/lay_plugins
cp -pd $bininstdir/pymod/klayout/*so ${pylibdir}
cp -pd $bininstdir/pymod/klayout/*py ${pylibdir}
for d in db tl rdb lib; do
  mkdir -p ${pylibdir}/$d
  cp -pd $bininstdir/pymod/klayout/$d/*py ${pylibdir}/$d
done

echo "Stripping..."
strip ${bindir}/*
strip ${pylibdir}/*.so
strip ${libdir}/db_plugins/*.so*
strip ${libdir}/lay_plugins/*.so*
```

---

On that note, is there anything I could be doing to minimize build times? I don't need to build the unit tests for example but I can't figure out if there's a flag or something to do so.

**Comment by donn** [comment]
I did mean MB, yeah, oops.

>  for example on CentOS 7, the release configuration still includes "-g", which bloats the object files

That would explain it. I know next to nothing about qmake.

I doubt it's the strip patterns personally- I carried them over 1:1 from your `makedeb.sh`, and attempting to run `strip` in the lib/ folder (the biggest culprit) nukes a number of the libraries in the folder that were present in the .deb packaging, so that was not it either.

---
